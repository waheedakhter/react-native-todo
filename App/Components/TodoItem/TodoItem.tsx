import { useNavigation } from "@react-navigation/native";
import * as React from "react";
import { Platform, Text, TouchableOpacity, View, ViewProps } from "react-native";
import SwiperComponent from "../SwiperComponent";
import { useDispatch } from "react-redux";
import colors from "../../Themes/Colors";
import style from "./TodoItemStyle";
import { TodosActions } from "../../Reducers/TodoReducer";

interface Props extends ViewProps {
  item: any;
  isActiveDelivery?: boolean;
}

const DeliveryItem: React.FunctionComponent<Props> = ({ item, isActiveDelivery }: Props) => {
  const dispatch = useDispatch();
  const navigation = useNavigation();
  const swipe = React.useRef(null);
  const removeFunctionality = React.useCallback(() => {
    dispatch(TodosActions.deleteTodoItem(item, {}));
  }, [item]);

  return (
    <SwiperComponent
      ref={swipe}
      onFullSwipe={() => {}}
      backView={
        <TouchableOpacity onPress={removeFunctionality} style={style.btnViewStyle}>
          <View style={style.deleteView}>
            <Text style={style.textStyle}>Remove</Text>
          </View>
        </TouchableOpacity>
      }
    >
      <TouchableOpacity onPress={() => navigation.navigate("TodoDetails", { isUpdate: true, item: item })}>
        {item && (
          <View
            style={[
              style.itemContainer,
              {
                shadowColor: isActiveDelivery
                  ? colors.red
                  : Platform.select({
                      android: colors.darkGrey,
                      ios: colors.lightGrey,
                    }),
              },
            ]}
          >
            <Text style={style.description} numberOfLines={1}>
              {item.description}
            </Text>
            <Text style={style.dateStyle}>{item.dateTime}</Text>
          </View>
        )}
      </TouchableOpacity>
    </SwiperComponent>
  );
};

export default DeliveryItem;
