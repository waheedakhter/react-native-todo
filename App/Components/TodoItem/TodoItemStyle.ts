import { StyleSheet } from "react-native";
import { ApplicationStyles, Colors } from "../../Themes";
import colors from "../../Themes/Colors";

export default StyleSheet.create({
  row: {
    flexDirection: "row",
    justifyContent: "space-between",
    paddingHorizontal: 20,
  },
  label: { flex: 1, color: colors.darkGrey },
  value: {
    flex: 2,
    textAlign: "right",
  },
  itemContainer: {
    borderRadius: 10,
    marginHorizontal: 10,
    marginTop: 10,
    padding: 10,
    ...ApplicationStyles.shadow,
  },
  description: { fontSize: 30 },
  dateStyle: { fontSize: 20, color: Colors.darkGrey },
  btnViewStyle: {
    backgroundColor: Colors.darkRed,
    flex: 1,
    justifyContent: "center",
    marginTop: 10,
  },
  deleteView: {
    justifyContent: "center",
    width: 100,
    alignItems: "center",
  },
  textStyle: {
    fontSize: 12,
    color: "white",
  },
});
