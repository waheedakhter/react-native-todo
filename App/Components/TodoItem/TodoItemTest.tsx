import * as React from "react";
import { Provider } from "react-redux";
import * as renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import TodoItem from "./TodoItem";

const mockStore = configureMockStore();
const store = mockStore({
  deliveries: {
    deliveries: [],
    activeDelivery: undefined,
    finishedDeliveries: {},
    fetching: false,
  },
});

describe("TodoItem UI Testing", () => {
  const component = renderer
    .create(
      <Provider store={store}>
        <TodoItem item={{ id: 1643380751002, description: "Item", dateTime: "Fri Jan 28 2022 18:35:08" }} />
      </Provider>,
    )
    .toJSON();
  it("renders snapshot as expected", () => {
    expect(component).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    expect(component).toBeTruthy();
  });
});
