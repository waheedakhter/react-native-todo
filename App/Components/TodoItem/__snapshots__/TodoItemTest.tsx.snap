// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem UI Testing renders snapshot as expected 1`] = `
<View>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "translateX": Array [
                Array [
                  AnimatedValue {
                    " __value": 0,
                  },
                  AnimatedValue {
                    " __value": -750,
                  },
                ],
              ],
            },
          ],
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "elevation": 10,
              "marginHorizontal": 10,
              "marginTop": 10,
              "padding": 10,
              "shadowColor": "#DDDDDD",
              "shadowOffset": Object {
                "height": 3,
                "width": 3,
              },
              "shadowOpacity": 1,
              "shadowRadius": 10,
            },
            Object {
              "shadowColor": "#DDDDDD",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "fontSize": 30,
            }
          }
        >
          Item
        </Text>
        <Text
          style={
            Object {
              "color": "#666666",
              "fontSize": 20,
            }
          }
        >
          Fri Jan 28 2022 18:35:08
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "translateX": AnimatedValue {
                " __value": 0,
              },
            },
          ],
          "width": 0,
        },
        Object {
          "height": "100%",
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#CE1141",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 12,
            }
          }
        >
          Remove
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(52, 52, 52, 0)",
        "height": "100%",
        "position": "absolute",
        "width": "6%",
      }
    }
  />
</View>
`;
